### DISCUSSION TITLE
Using handle() for a Pingdom provider.

### DISCUSSION THREAD

**Discussion by leoalmeidadotme** [hotmeteor/receiver#24] [Receivers]
Hi there!

Laravel newbie here. I'm trying to receive a webhook from Pingdom. For example, when a website goes down.

I managed to create a new provider, verify() is working well (checking one of the JSON for a certain hash). But I can’t use the handle() to register the incoming hook/JSON as a new notice/update.

I would like to get the JSON and register on my database something such as:

- This is a test message triggered by a user in My Pingdom" [long_description]
- Created at 2023-04-27T09:14:53 [state_changed_utc_time]
- website.io [hostname]
- Proin ipsum sapien sed PBR vivamus sagittis curabitur ligula brunch metus sapien nulla arcu indie vulputate elementum metus. Eget hoodie porta congue malesuada non tofu leo duis pellentesque lectus brunch adipiscing sagittis donec morbi San Francisco ipsum maecenas ut. [custom_message]

I'm, trying as per the documented example - But nothing is registered on the logs, as a starting point. Any ideas, please?

```
<?php

namespace App\Http\Handlers\Pingdom;

use Illuminate\Foundation\Bus\Dispatchable;

class PingdomHandler
{
    use Dispatchable;
    
    public function __construct(public string $event, public array $data)
    {
    }

    public function handle()
    {
        $data = $this->getData();
        $event = $this->getEvent();
        
        logger("Received Pingdom event: {$event}");
        logger("Event data: " . json_encode($data));
    }
}

```

**Comment by hotmeteor** [comment]
Hey @leoalmeidadotme, sorry for the late reply. Are your events being queued and do you have your queue workers running? If you don't then these handlers will never be executed.

**Comment by leoalmeidadotme** [comment]
No worries, thanks for replying @hotmeteor . After trying for two weeks without success I rolled back to implementing the webhooks manually - as a newbie, I can have a clue about what I'm doing ;) Cheers.

**Comment by hotmeteor** [comment]
Do you know what the issue was? I'd love to be able to improve the Receiver package to better handle these situations.

**Comment by leoalmeidadotme** [comment]
Unfortunately I don't remember the issue, Adam.
As feedback to improve Receiver, I would recommend adding more examples (beyond the Stripe ones, for example) so non-pro coders could make the most of the package.
